clientconn.go

```diff
@@ -1981,16 +1981,14 @@ func (cc *ClientConn) determineAuthority() error {
 	}
 
 	endpoint := cc.parsedTarget.Endpoint()
-	target := cc.target
+	auth, isAuthOverrider := any(cc.resolverBuilder).(resolver.AuthorityOverrider)
 	switch {
```

Comment:
Optional: maybe just convert this into a few `if`s and do the auth overrider as a compound `if`.